src/actions/vstudio/vs2010_nuget.lua

```diff
@@ -54,33 +37,148 @@
 		end
 	end
 
+	function nuget2010.packageAPIInfo(prj, package)
+		local id = nuget2010.packageId(package)
+		local version = nuget2010.packageVersion(package)
 
---
--- Generates the packages.config file.
---
+		if not packageAPIInfos[package] then
+			local packageAPIInfo = {}
 
-	function nuget2010.generatePackagesConfig(obj)
-		local wks = obj.workspace
+			printf("Examining NuGet package '%s'...", id)
+			io.flush()
 
-		local done = {}
-		local packages = {}
-		for prj in p.workspace.eachproject(wks) do
-			for i = 1, #prj.nuget do
-				local package = prj.nuget[i]
+			local response, err, code = http.get(string.format("https://api.nuget.org/v3/registration1/%s/index.json", id:lower()))
```

Comment:
I did some digging and I think that reporting the file listing is unsupported by [NuGet.Server](https://github.com/NuGet/NuGet.Server). I think they have a different server implementation running at `nuget.org` and they populate the file listings there with a background task.

If I'm right about this, we could support custom package sources, but people wouldn't be able to use any servers with it, because the open source server doesn't return the data that we need...

I suppose people could use it with C++ projects because we don't need the file listing for those, but C# projects wouldn't work with it.

One thing that we could do is file an issue to the [NuGet.Server issue tracker](https://github.com/nuget/NuGetGallery/issues?q=is%3Aissue+label%3A%22Area%3A+NuGet.Server%22+is%3Aopen), but I'm not comfortable doing so until I'm sure that I've understood the issue correctly.